### PR TITLE
LPS-199931 | Master

### PR DESCRIPTION
**Pull request by GergoMathe**
[LPS-199931](https://liferay.atlassian.net/browse/LPS-199931)

### CODE REVIEW

**Comment by GergoMathe** [comment]
ci:test:sf

**Comment by GergoMathe** [comment]
ci:test:security

**Comment by GergoMathe** [comment]
ci:test:passwordpolicies:nocompile

**Comment by GergoMathe** [comment]
ci:test:password-policies:nocompile

**Comment by GergoMathe** [comment]
ci:test:sf

**Comment by GergoMathe** [comment]
ci:test:password-policies:nocompile

**Review by gaborlovasdotcom — APPROVED** [review]

**Review by gaborlovasdotcom — CHANGES_REQUESTED** [review]
Could you please fix the failing TCs?

**Comment by JoyceWang08211** [comment]
Hi @GergoMathe, test cases failed at the set-up part because the property passwords.default.policy.change.required=true. If you use User.firstLoginPG();, then you need to reset the password during the first login. You need to bypass this reset part.
cc @gaborlovasdotcom

**Comment by GergoMathe** [comment]
ci:test:sf

**Comment by GergoMathe** [comment]
ci:test:password-policies:nocompile

**Comment by GergoMathe** [comment]
ci:test:sf

**Comment by GergoMathe** [comment]
ci:test:sf

**Comment by GergoMathe** [comment]
ci:test:password-policies:nocompile

**Comment by GergoMathe** [comment]
ci:test:sf

**Comment by GergoMathe** [comment]
ci:test:password-policies:nocompile

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/passwordpolicies/PasswordPolicies.testcase [hunk 1]
```diff
@@ -541,6 +541,271 @@ definition {
 			userEmailAddress = "userea@liferay.com");
 	}
 
+	@description = "This is a use case for LPS-193886 (TC-2). A new User cannot set their new password through the password setup link, if it is expired."
+	@priority = 5
+	test NewUserCannotSetNewPasswordThroughTheExpiredLink {
+		property custom.properties = "feature.flag.LPS-193884=true${line.separator}passwords.default.policy.change.required=false${line.separator}passwords.default.policy.reset.ticket.max.age=300";
```
Comment (JoyceWang08211) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/passwordpolicies/PasswordPolicies.testcase:547]: There is no need to add the _passwords.default.policy.change.required=false_ property as its default value is false:
https://github.com/liferay/liferay-portal/blob/master/build-test.xml#L11911
The following cases also have the same problem.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/passwordpolicies/PasswordPolicies.testcase [hunk 2]
```diff
@@ -541,6 +541,271 @@ definition {
...
+				userEmailAddress = "b@liferay.com",
+				userFirstName = "b",
+				userLastName = "b",
+				userScreenName = "b");
+
+			User.logoutPG();
+		}
+
+		task ("When: The link in the received email is opened after it expired (wait for it to expire)") {
+			Pause(value1 = 310000);
```
Comment (JoyceWang08211) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/passwordpolicies/PasswordPolicies.testcase:578]: We need to add _pause_ very carefully. 
Please try to see if you can reduce the pause time by shortening the expiration time(aka passwords.default.policy.reset.ticket.max.age=300). 
If not, I'm okay with the long pause.
In addition, when adding a pause, we need to add a description to explain the necessity of this pause. Please refer to:
https://github.com/liferay/liferay-portal/blob/master/portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/oauth2/OAuth2.testcase#L976-L978

**Review by JoyceWang08211 — CHANGES_REQUESTED** [review]
See comments in PR

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/passwordpolicies/PasswordPolicies.testcase (hunk 1 above)
Comment (GergoMathe) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/passwordpolicies/PasswordPolicies.testcase:547]: I'll remove that custom property as it is and do a rerun.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/passwordpolicies/PasswordPolicies.testcase (hunk 2 above)
Comment (GergoMathe) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/passwordpolicies/PasswordPolicies.testcase:578]: Sadly, there's no way to work around this one as we are just waiting for the setup link to expire and it has no visible feeedback. I'll add the explanation to the test cases.

**Comment by GergoMathe** [comment]
ci:test:sf

**Comment by GergoMathe** [comment]
ci:test:password-policies:nocompile

**Comment by GergoMathe** [comment]
ci:test:sf

**Comment by GergoMathe** [comment]
ci:test:password-policies:nocompile

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/passwordpolicies/PasswordPolicies.testcase [hunk 3]
```diff
@@ -541,6 +541,271 @@ definition {
...
+			PasswordPolicies.openPasswordPoliciesAdmin();
+
+			PasswordPolicies.assignMembers(
+				assigneeName = "d d",
+				assigneeType = "Users",
+				passwordPolicyName = "New Policy");
+
+			User.logoutPG();
+
+			Pause(value1 = 310000);
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/passwordpolicies/PasswordPolicies.testcase:777]: Could you please rework the Pause statements?

**Review by gaborlovasdotcom — CHANGES_REQUESTED** [review]

**Comment by zsigmondrab** [comment]
Merged

**Pull request closed without merging**